modules/util/css-builder/src/main/java/com/liferay/css/builder/CSSBuilder.java

```diff
@@ -556,14 +591,71 @@ private void _writeOutputFile(String fileName, String content, boolean rtl)
 
 	private static RTLCSSConverter _rtlCSSConverter;
 
-	private final boolean _appendCssImportTimestamps;
-	private final boolean _cleanPortalCommonDir;
-	private final String _docrootDirName;
-	private final boolean _generateSourceMap;
-	private final String _outputDirName;
-	private final String _portalCommonDirName;
-	private final int _precision;
-	private final Pattern[] _rtlExcludedPathPatterns;
+	@Parameter(
+		description = "Whether to append the current timestamp to the URLs in the @import CSS at-rules.",
+		names = "sass.append.css.import.timestamps"
+	)
+	private boolean _appendCssImportTimestamps =
+		CSSBuilderArgs.APPEND_CSS_IMPORT_TIMESTAMPS;
+
+	private boolean _cleanPortalCommonDir;
+
+	@Parameter(
+		description = "If the java plugin is applied: The first resources directory of the main source set (by default: src/main/resources).\nIf the war plugin is applied: project.webAppDir.\nOtherwise: null",
+		names = "sass.docroot.dir"
+	)
+	private String _docrootDirName = CSSBuilderArgs.DOCROOT_DIR_NAME;
+
+	@Parameter(
+		description = "Whether to generate source maps for easier debugging.",
+		names = "sass.generate.source.map"
+	)
+	private boolean _generateSourceMap;
+
+	@Parameter(
+		description = "The name of the sub-directories where the SCSS files are compiled to. " +
+			"For each directory that contains SCSS files, a sub-directory with this name is created. ",
+		names = "sass.output.dir"
+	)
+	private String _outputDirName = CSSBuilderArgs.OUTPUT_DIR_NAME;
+
+	@Parameter
+	private List<String> _parameters = new ArrayList<>();
+
+	private String _portalCommonDirName;
+
+	@Parameter(
+		description = "The value of the portalCommonDir property if set; otherwise portalCommonFile.",
+		names = {"sass.portal.common.path", "sass.portal.common.dir"}
+	)
+	private String _portalCommonPath;
+
+	@Parameter(
+		description = "The numeric precision of numbers in Sass.",
+		names = "sass.precision"
+	)
+	private Integer _precision = CSSBuilderArgs.PRECISION;
+
+	private Pattern[] _rtlExcludedPathPatterns;
+
+	@Parameter(
+		description = "The SCSS file patterns to exclude when converting for right-to-left (RTL) support.",
+		names = "sass.rtl.excluded.path.regexps"
+	)
+	private String _rtlExcludedPathRegexps;
+
 	private SassCompiler _sassCompiler;
 
+	@Parameter(
+		description = "The type of Sass compiler to use. Supported values are \"jni\" and \"ruby\". If not set, defaults to \"jni\".",
+		names = "sass.compiler.class.name"
+	)
+	private String _sassCompilerClassName;
+
+	@Parameter(
+		description = "The name of the directories, relative to docrootDir, which contain the SCSS files to compile. All sub-directories are searched for SCSS files as well.",
+		names = "sass.dir"
+	)
+	private List<String> _sassDirs = new ArrayList<>();
```

Comment:
You should see what happens if you ask for a `List<File>` instead. I think it would work.